tds.go

```diff
@@ -850,6 +850,13 @@ func dialConnection(ctx context.Context, c *Connector, p connectParams) (conn ne
 	if ip == nil {
 		ips, err = net.LookupIP(p.host)
```

Comment:
I can think of two problematic scenarios here:
- Some companies (grafana included) use Public DNS providers to resolve Private IPs.
- Some companies have both a Public DNS and a Private DNS. This is for example the setup we have when relying on Private Link.

I both scenarios, the DNS lookup will succeed and the Dialer will not be called. 

I suggest the Dialer check happens before the `ip := net.ParseIP(p.host)`

Reply:
Really good catch! I updated the code accordingly.

Reply:
Thinking about this further, I think we just need to move it up to under if `ip==nil` (before we try to resolve the dns), rather than before we try to parse the ip. Because the dialer will still be used during the ip lookups, but we still want to preserve the functionality where multiple ips are parsed.

Reply:
I think it's the same isn't it ? The multiple IP will only be valid if resolving a hostname returns multiple IPs.

Reply:
🤦‍♀️ you're right - I was thinking a list of ips could be specified, but from the `else` statement, it can't be that way. So yes, it's the same.